pyproject.toml

```
@@ -1,15 +1,193 @@
+[project]
+name = "photutils"
+description = "An Astropy package for source detection and photometry"
+requires-python = ">=3.8"
+authors = [
+    { name = "Photutils Developers", email = "photutils.team@gmail.com" },
+]
+keywords = [
+    "astronomy",
+    "astrophysics",
+    "photometry",
+    "aperture",
+    "psf",
+    "source detection",
+    "background",
+    "segmentation",
+    "centroids",
+    "isophote",
+    "morphology",
+]
+classifiers = [
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Natural Language :: English",
+    "Operating System :: OS Independent",
+    "Programming Language :: Cython",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Topic :: Scientific/Engineering :: Astronomy",
+]
+dependencies = [
+    "numpy>=1.22",
+    "astropy>=5.0",
+]
+dynamic = [
+    "version",
+]
+
+[project.readme]
+file = "README.rst"
+content-type = "text/x-rst"
+
+[project.license]
+file = "LICENSE.rst"
+content-type = "text/plain"
+
+[project.urls]
+Homepage = "https://github.com/astropy/photutils"
+
+[project.optional-dependencies]
+all = [
+    "scipy>=1.7.0",
+    "matplotlib>=3.5.0",
+    "scikit-image>=0.19.0",
+    "scikit-learn>=1.0",
+    "gwcs>=0.18",
+    "bottleneck",
+    "tqdm",
+    "rasterio",
+    "shapely",
+]
+test = [
+    "pytest-astropy",
+]
+docs = [
+    "sphinx",
+    "sphinx-astropy>=1.6",
+    "scipy>=1.7.0",
+    "matplotlib>=3.5.0",
+    "scikit-image>=0.19.0",
+    "scikit-learn>=1.0",
+    "gwcs>=0.18",
+    "rasterio",
+    "shapely",
+    "tomli; \"python-version\" < \"3.11\"",
+]
+
 [build-system]
-requires = ['setuptools',
-            'setuptools_scm>=6.2',
-            'cython>=0.29.30',
-            'oldest-supported-numpy',
-            'extension-helpers']
-build-backend = 'setuptools.build_meta'
+requires = [
+    "setuptools>=61.2",
+    "setuptools_scm[toml]>=3.4",
+    "wheel",
+    "cython>=0.29.30",
+    "oldest-supported-numpy",
+    "extension-helpers",
+]
+build-backend = "setuptools.build_meta"
 
 [tool.setuptools_scm]
 write_to = "photutils/version.py"
+write_to_template = """
+# Note that we need to fall back to the hard-coded version if either
+# setuptools_scm can't be imported or setuptools_scm can't determine the
+# version, so we catch the generic 'Exception'.
+try:
+    from setuptools_scm import get_version
+    version = get_version(root='..', relative_to=__file__)
+except Exception:
+    version = '{version}'
+"""
+
+[tool.setuptools]
+zip-safe = false
+include-package-data = false
+license-files = [
+    "LICENSE.rst",
+]
+
+[tool.setuptools.packages.find]
+namespaces = false
+
+[tool.setuptools.package-data]
+photutils = [
+    "CITATION.rst",
+]
+"photutils.datasets" = [
+    "data/*",
+]
+"photutils.detection.tests" = [
+    "data/*",
+]
+"photutils.isophote.tests" = [
+    "data/*",
+]
```

```suggestion
]
"photutils.psf.tests" = [
    "data/*",
]
```

This was added after your initial draft PR.